_delphi_utils_python/delphi_utils/validator/utils.py

```diff
@@ -77,3 +77,47 @@ def aggregate_frames(frames_list):
         all_frames.append(df)
 
     return pd.concat(all_frames).reset_index(drop=True)
+
+def lag_converter(lag_dict):
+    """Convert a dictionary of lag values into the proper format.
+
+    Parameters
+    ----------
+    lag_dict: Dict[str, str or int]
+        Keys are either 'all', or signal names
+        Values are either numeric or days of the week, represented by
+            "Sunday+[0-7],[0-9]", first part represents the day of the week
+            the upload happens, while the second number represents lag upon
+            upload.
+    Returns
+    ----------
+    Dict[str, int]
+        Keys are all active signal names for an indicator, or "all"
+        Values are obtained by looking at signal name entries,
+            If not present then by the 'all' indicator,
+            else 1.
+    """
+    def value_interpret(value):
+        """Convert value from string to numeric, including sunday+m,n."""
+        if isinstance(value, int):
+            value_num = value
+        elif value.startswith("sunday+"):
+            # Check that the number proceeding sunday+ has length of 1
+            assert value[8] == ","
+            # Value_num calculates the number of days between today and the
+            # Update day, except that if both days of the week are the same,
+            # Expected lag is 7 (in case updates aren't in yet)
+            value_num = (date.today().isoweekday() - int(value[7:8]) - 1) % 7 + 1
+            # Add on expected lag to lag from weekdays
+            value_num += int(value[9:])
+        else:
+            value_num = int(value)
+        return value_num
+
+    # Add 'all' to lag_dict keys, give it default value of 1
+    lag_dict["all"] = lag_dict.get("all", "1")
```

Comment:
I'm not understanding the purpose of this bit. Above, like in `dynamic.py`, we're calling `self.params.min_expected_lag.get('all', 1)` and `self.params.max_expected_lag.get('all', 10)`, so we're already providing defaults for the "all" key. Also, you mentioned above that you wanted to use a larger value for max_lag, but here min and max lags are always going to be the same.

Reply:
Hm, this is definitely a bit of redundancy. Would you prefer I remove it then? If yes, should I remove this line and keep the 1/10?

Reply:
> should I remove this line and keep the 1/10?

Yes, I think it makes more sense to remove it here and fill in as needed in other parts of the code because: 1) you wanted to use different "all" defaults for min vs max lag and this doesn't allow that, and 2) we don't always need to have the "all" key.